src/node_trace_events.cc

```diff
@@ -70,20 +70,20 @@ static void Emit(const FunctionCallbackInfo<Value>& args) {
     id = args[3]->IntegerValue(context).ToChecked();
   }
 
-  // TODO(AndreasMadsen): Two extra arguments are not supported.
   // TODO(AndreasMadsen): String values are not supported.
   int32_t num_args = 0;
-  const char* arg_names[1];
-  uint8_t arg_types[1];
-  uint64_t arg_values[1];
+  const char* arg_names[2];
+  uint8_t arg_types[2];
+  uint64_t arg_values[2];
 
   // Create Utf8Value in the function scope to prevent deallocation.
   // The c-string will be copied by TRACE_EVENT_API_ADD_TRACE_EVENT at the end.
   Utf8Value arg1NameValue(env->isolate(), args[4]);
+  Utf8Value arg2NameValue(env->isolate(), args[6]);
 
-  if (args.Length() < 6 || (args[4]->IsUndefined() && args[5]->IsUndefined())) {
-    num_args = 0;
-  } else {
+  if (args.Length() >= 6 &&
+      !args[4]->IsUndefined() &&
```

Comment:
Was the logic wrong here before? If we're not changing the logic, the negation of the statement should be 
`if(args.Length() >= 6 && (!args[4]->IsUndefined() || !args[5]->IsUndefined()))`. DeMorgan changes the second `&&` to a `||`.

Reply:
Right. I think it is a matter of opinion. But since we have validation after we should properly go with what you suggest.